packages/valory/skills/decision_maker_abci/behaviours/sampling.py

```diff
@@ -99,15 +99,49 @@ def processable_bet(self, bet: Bet) -> bool:
 
     def _sampled_bet_idx(self, bets: List[Bet]) -> int:
         """
-        Sample a bet and return its id.
-
-        The sampling logic is relatively simple at the moment.
-        It simply selects the unprocessed bet with the largest liquidity.
+        Sample a bet and return its index.
+
+        The sampling logic follows the specified priority logic:
+        1. Filter out all the bets that have a processed_timestamp != 0 to get a list of new bets.
+        2. If the list of new bets is not empty:
+           2.1 Order the list in decreasing order of liquidity (highest liquidity first).
+           2.2 For bets with the same liquidity, order them in decreasing order of market closing time (openingTimestamp).
+        3. If the list of new bets is empty:
+           3.1 Order the bets in decreasing order of invested_amount.
+           3.2 For bets with the same invested_amount, order them in increasing order of processed_timestamp (least recently processed first).
+           3.3 For bets with the same invested_amount and processed_timestamp, order them in decreasing order of liquidity.
+           3.4 For bets with the same invested_amount, processed_timestamp, and liquidity, order them in decreasing order of market closing time (openingTimestamp).
 
         :param bets: the bets' values to compare for the sampling.
-        :return: the id of the sampled bet, out of all the available bets, not only the given ones.
+        :return: the index of the sampled bet, out of all the available bets, not only the given ones.
         """
-        return self.bets.index(max(bets))
+
+        # Filter out all the best with the smallest queue number
+        least_queue_number = min([bet.queue_no for bet in bets])
+        priority_bets = [bet for bet in bets if bet.queue_no == least_queue_number]
+
+        # Filter out all the bets that have a processed_timestamp == 0
+        new_in_priority_bets = [bet for bet in bets if bet.processed_timestamp == 0]
+
+        if new_in_priority_bets:
```

Comment:
Here we always give priority to the new bets arriving. However, the spec says that priority to new bets should only be done on startup, i.e., `current_queue_number == 0`